Collector/Formatter.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Http\HttplugBundle\Collector;
+
+use Exception;
+use Http\Client\Exception\HttpException;
+use Http\Client\Exception\TransferException;
+use Http\Message\Formatter as FormatterInterface;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * @author Fabien Bourigault <bourigaultfabien@gmail.com>
+ *
+ * @internal
+ */
+class Formatter implements FormatterInterface
+{
+    /**
+     * @var FormatterInterface
+     */
+    private $formatter;
+
+    /**
+     * @param FormatterInterface $formatter
+     */
+    public function __construct(FormatterInterface $formatter)
+    {
+        $this->formatter = $formatter;
+    }
+
+    /**
+     * Formats an exception.
+     *
+     * @param Exception $exception
+     *
+     * @return string
+     */
+    public function formatException(Exception $exception)
+    {
+        if ($exception instanceof HttpException) {
+            return $this->formatter->formatResponse($exception->getResponse());
+        }
+
+        if ($exception instanceof TransferException) {
+            return $exception->getMessage();
```

Comment:
should we add some sort of prompt? message alone could be incomplete or even empty... "Request failed: " maybe?